packaging/CPackDebUploadPPA.cmake

```diff
@@ -43,10 +43,12 @@ if(NOT CPACK_DEBIAN_GPG_RET EQUAL "0")
 endif()
 
 # hack to advance the version from the legacy version like this:
-# dpkg --compare-versions 2.4.0~alpha~pre1~git7859  lt 2.4.0~alpha1~5463~gf2da9e619d && echo true
-# dpkg --compare-versions 2.4.0~alpha1~5463~gf2da9e619d lt 2.4.0 && echo true
+# dpkg --compare-versions 2.4.0~alpha~pre1~git7859  lt 2.4.0~alpha2~5463~gf2da9e619d && echo true
+# and from changing to --first-parent
+# dpkg --compare-versions 2.4.0~alpha1~6372~g0244af2e04-1~impish lt 2.4.0~alpha2~805~g4f13bc1d5d-1~impish && echo true
+# dpkg --compare-versions 2.4.0~alpha2~5463~gf2da9e619d lt 2.4.0 && echo true
 if(DEB_UPLOAD_PPA MATCHES "nightlies")
-  string(REPLACE "2.4~alpha~" "2.4.0~alpha1~" CPACK_DEBIAN_PACKAGE_VERSION "${CPACK_DEBIAN_PACKAGE_VERSION}")
+  string(REPLACE "2.4~alpha~" "2.4.0~alpha2~" CPACK_DEBIAN_PACKAGE_VERSION "${CPACK_DEBIAN_PACKAGE_VERSION}")
```

Comment:
Do we need to use alpha2 for 2.5.0 also or is this a temporary fix? Does this need to be documented somehow?

Reply:
No, this is just a temporary workaround until the transition to the git describe version is finished.
That's why the Replace statement works only for "2.4~alpha~"
 
2.5.0~alpha will stay unchanged and will work. See:  
```
dpkg --compare-versions 2.4.0~alpha2~5463~gf2da9e619d lt 2.5.0~alpha && echo true
true
```

Reply:
Ok, the comment say "hack". I won't touch it and hopefully no one else needs to.